proposals/fate-operator-proposal.md

```diff
@@ -0,0 +1,199 @@
+
+-   [Background](#background)
+-   [Motivation](#motivation)
+-   [Goals](#goals)
+-   [Non-Goals](#non-goals)
+-   [Design](#design)
+    -   [Container images and deploying FATE cluster on
+        Kubernetes](#container-images-and-deploying-fate-cluster-on-kubernetes)
+    -   [Custom Resource Definition](#custom-resource-definition)
+        -   [Kubefate](#kubefate)
+        -   [FATECluster](#fatecluster)
+        -   [FateJob](#fatejob)
+    -   [Controller design](#controller-design)
+
+_Status_
+* 2020-5-14 – Draft v1
+
+## Background
+Federated machine learning (FML) is a machine learning setting where many clients (e.g. mobile devices or organizations) collaboratively train a model under the coordination of a central server while keeping the training data decentralized. Only the encrypted mediate parameters are exchanged between clients with MPC or homomorphic encryption.
+![Federated Machine Learning](diagrams/fate-operator-fl.png)
+
+FML has received significant interest recently, because of its effectiveness to solve data silos and data privacy preserving problems. Companies participated in federated machine learning include 4Paradigm, ANT Financial, Data Republic, Google, Huawei, Intel, JD.com, Microsoft, Nvidia, OpenMind, Pingan Technology, Sharemind, Tencent, VMware, Webank etc.
```

Comment:
What do you mean by “participated” here?

Reply:
"Participant" in here means: join the research, and at least have own FML framework or wrapped existing frameworks as a solution. But not includes the companies just use FML or joined training. e.g. Googgle has Tensorflow Federated, OpenMide has PySyft, Tencent has solution based on FATE, Nvidia has Clara...

Reply:
Given that there are so many frameworks there, would it possible to define some interface that can be generalized across federated learning frameworks?

Reply:
Yeah, same idea here. Could we have something like fml-common?

Reply:
Indeed good idea. But FML is new, and there are many framework with different design purples. We would like to work with them and provide a common APIs. This is also why we hope to join Kubeflow community, that we can work together with other FML teams.